src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderManager.cs

```diff
@@ -26,8 +26,6 @@ public DicomInstanceEntryReaderManager(IEnumerable<IDicomInstanceEntryReader> di
     /// <inheritdoc />
     public IDicomInstanceEntryReader FindReader(string contentType)
     {
-        EnsureArg.IsNotNull(contentType, nameof(contentType));
```

Comment:
We should probably keep this check in place while we implement the error handling.

Reply:
We don't necessarily need this. Given that the reader returns null if `ContentType` is null, we are good. The caller is responsible for making sure that the reader is not null. In our case, we always check the reader and return `UnsupportedMediaType` status code if the reader is not found.
Please do let me know if I misunderstood the comment.

Reply:
I have also updated the `StoreHandler` tests to verify that we return `UnsupportedMediaType` exception when `ContentType` is null.